### ISSUE TITLE
stickynoteenc traceback

### ISSUE THREAD

**Issue by LewisNeal** [leo-editor/leo-editor#1265]
I have pycryptodome package installed. Running stickynoteenc command launches the GUI and asks you to enter a key. Pressing OK button gives errors.

Leo Log Window
Leo 6.0-b2-devel, devel branch, build 2397655ff2
2019-07-26 10:01:19 -0500
Python 3.6.8, PyQt version 5.13.0
Windows 10 AMD64 (build 10.0.17134) SP0
Traceback (most recent call last):

  File "N:\git\leo-editor\leo\core\leoCommands.py", line 2078, in executeAnyCommand
return command(event)

  File "N:\git\leo-editor\leo\plugins\stickynotes.py", line 233, in stickynoteenc_f
decoded = sn_decode(v.b)

  File "N:\git\leo-editor\leo\plugins\stickynotes.py", line 244, in sn_decode
return AES.new(__ENCKEY[0]).decrypt(base64.b64decode(s)).decode('utf-8').strip()

TypeError: new() missing 1 required positional argument: 'mode'

**Comment by edreamleo** [comment]
Rev 8e03aa1 adds the required AES.MODE_EAX mode.  This should work with recent versions of the Crypto.Cipher package.  Please reopen if other problems surface.

**Comment by edreamleo** [comment]
But now there are padding issues, which I don't understand:
```
 File "c:\leo.repo\leo-editor\leo\core\leoCommands.py", line 2078, in executeAnyCommand
    return command(event)
  File "c:\leo.repo\leo-editor\leo\plugins\stickynotes.py", line 233, in stickynoteenc_f
    decoded = sn_decode(v.b)
  File "c:\leo.repo\leo-editor\leo\plugins\stickynotes.py", line 246, in sn_decode
    return AES.new(__ENCKEY[0], AES.MODE_EAX).decrypt(base64.b64decode(s)).decode('utf-8').strip()
  File "c:\Anaconda3\lib\base64.py", line 87, in b64decode
    return binascii.a2b_base64(s)
binascii.Error: Incorrect padding
``

**Comment by LewisNeal** [comment]
GUI launches, I can now create an encrypted node. However switching to another node crashes Leo.

Leo 6.0-b2-devel, devel branch, build 8e03aa1e13
2019-07-31 05:04:21 -0500
rss.py: can not import feedparser
setHomeDirectory.py g.app.homeDir set to c:\
wrote recent file: C:/Users/lewis/.leo/.leoRecentFiles.txt
Traceback (most recent call last):
  File "N:\git\leo-editor\leo\plugins\stickynotes.py", line 295, in f
    newfunc()
  File "N:\git\leo-editor\leo\plugins\stickynotes.py", line 201, in focusin
    decoded = sn_decode(v.b)
  File "N:\git\leo-editor\leo\plugins\stickynotes.py", line 244, in sn_decode
    return AES.new(__ENCKEY[0], AES.MODE_EAX).decrypt(base64.b64decode(s)).decode('utf-8').strip()
  File "C:\Users\lewis\AppData\Local\Programs\Python\Python37\lib\site-packages\Crypto\Cipher\AES.py", line 232, in
new
    return _create_cipher(sys.modules[__name__], key, mode, *args, **kwargs)
  File "C:\Users\lewis\AppData\Local\Programs\Python\Python37\lib\site-packages\Crypto\Cipher\__init__.py", line 79, in _create_cipher
    return modes[mode](factory, **kwargs)
  File "C:\Users\lewis\AppData\Local\Programs\Python\Python37\lib\site-packages\Crypto\Cipher\_mode_eax.py", line 408, in _create_eax_cipher
    return EaxMode(factory, key, nonce, mac_len, kwargs)
  File "C:\Users\lewis\AppData\Local\Programs\Python\Python37\lib\site-packages\Crypto\Cipher\_mode_eax.py", line 112, in __init__
    for i in range(0, 3)
  File "C:\Users\lewis\AppData\Local\Programs\Python\Python37\lib\site-packages\Crypto\Cipher\_mode_eax.py", line 112, in <listcomp>
    for i in range(0, 3)
  File "C:\Users\lewis\AppData\Local\Programs\Python\Python37\lib\site-packages\Crypto\Hash\CMAC.py", line 305, in new
    update_after_digest)
  File "C:\Users\lewis\AppData\Local\Programs\Python\Python37\lib\site-packages\Crypto\Hash\CMAC.py", line 59, in __init__
    self._key = _copy_bytes(None, None, key)
  File "C:\Users\lewis\AppData\Local\Programs\Python\Python37\lib\site-packages\Crypto\Util\py3compat.py", line 156, in _copy_bytes
    return seq[start:end]
TypeError: 'NoneType' object is not subscriptable
[end]

**Comment by edreamleo** [comment]
Rev bf259da in devel catches all exceptions in sn_decode, so (presumably) Leo won't crash even if encryption doesn't work.

**Comment by vitalije** [comment]
Fixed in cfd90673e

**Comment by edreamleo** [comment]
On Wed, Jul 31, 2019 at 10:41 AM Виталије Милошевић <
notifications@github.com> wrote:

> Fixed in cfd9067
> <https://github.com/leo-editor/leo-editor/commit/cfd90673e467f3d2117e043d7b6b28748e81751a>
>

Thanks for this.  I appreciate it.

Edward
